jsonsch/init.go

```diff
@@ -120,7 +128,125 @@ func appendTuple(tup tuple, dstSlice []interface{}, key string) ([]interface{},
 		if err != nil {
 			return nil, err
 		}
-		childInst, err := InitSchema(childSchema, true)
+		childInst, err := InitSchema(childSchema, true, false)
+		if err != nil {
+			return nil, err
+		}
+		elem = childInst
+
+	case Array:
+		elem = make([]interface{}, 0)
+		elemSchema, ok := tup.Data["items"]
+		if !ok {
+			return nil, fmt.Errorf("key '%v' has type 'Array' but no 'items' field", key)
+		}
+		childTup, err := prop2Tuple(elemSchema, fmt.Sprintf("%v.items", key))
+		if err != nil {
+			return nil, err
+		}
+		childSlice, ok := elem.([]interface{})
+		if !ok {
+			return nil, fmt.Errorf("type assertion failed in Array case")
+		}
+		newSlice, err := appendTuple(childTup, childSlice, key)
+		if err != nil {
+			return nil, err
+		}
+		elem = newSlice
+
+	default:
+		return nil, fmt.Errorf("element of array at key '%v' has unrecognized 'type' field '%v'", key, tup.Type)
+	}
+	return append(dstSlice, elem), nil
+}
+
+func generateRandomString(length int) string {
+	rand.Seed(time.Now().Unix())
+
+	var letters = []rune("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ")
+
+	randomString := make([]rune, length%2147483647)
```

Comment:
I think we should change 2147483647 to a smaller number. We can hard-code it as like 64 or 128 for now and then make it into a flag later on.